src/main/java/redis/clients/jedis/BinaryShardedJedis.java

```diff
@@ -41,14 +45,19 @@ public void disconnect() {
     for (Jedis jedis : getAllShards()) {
       if (jedis.isConnected()) {
         try {
-          jedis.quit();
+          // need a proper test, probably with mock
+          if (!jedis.isBroken()) {
+            jedis.quit();
+          }
         } catch (JedisConnectionException e) {
           // ignore the exception node, so that all other normal nodes can release all connections.
+          logger.error("Error while QUIT", e);
         }
         try {
           jedis.disconnect();
```

Comment:
Can we actually disconnect in a broken state?

Reply:
Yes, we can. It actually flushes socket stream and closes socket.

Reply:
But wouldn't flush fail in that case? (I guess the `finally` in `Connection.disconnet` still makes sure the socket is closed if possible so it might be fine), if your test and manual run works fine then that's okay then.

Reply:
Jedis is disconnecting broken connections since ages. It is yet to be pin-pointed as an issue :laughing:

Reply:
I yield to empirical evidence then 😅